shutdown/legacy.go

```diff
@@ -0,0 +1,84 @@
+package shutdown
+
+import (
+	"github.com/ThomasObenaus/go-base/shutdown/health"
+	"github.com/ThomasObenaus/go-base/shutdown/log"
+	"github.com/ThomasObenaus/go-base/shutdown/signal"
+	"github.com/ThomasObenaus/go-base/shutdown/stop"
+	"github.com/rs/zerolog"
+)
+
+type ShutdownHandler struct {
+	stoppableItems stopIF
+	signalHandler  signalHandlerIF
+	log            logIF
+	health         healthIF
+}
+
+func InstallHandler(orderedStopables []stop.Stoppable, logger zerolog.Logger) *ShutdownHandler {
+	shutdownHandler := &ShutdownHandler{
+		stoppableItems: &stop.OrderedStoppableList{},
+		log:            log.ShutdownLog{Logger: logger},
+		health:         &health.Health{},
+	}
+
+	for _, stopable := range orderedStopables {
+		err := shutdownHandler.stoppableItems.AddToBack(stopable)
+		if err != nil {
+			logger.Error().Err(err).Msgf("unexpected error adding stoppable to internal list")
+			return nil
+		}
+	}
+
+	handler := signal.NewDefaultSignalHandler(shutdownHandler)
+	shutdownHandler.signalHandler = handler
+
+	return shutdownHandler
+}
+
+func (h *ShutdownHandler) Register(stoppable stop.Stoppable, front ...bool) {
+	addToBack := isFirstBoolUndefinedOrFalse(front)
```

Comment:
a easy way to avoid that register is called while a shutdown is currently running would be 

```
func Register(...) error {
	h.mux.Lock()
	defer h.mux.Unlock()
      
if h.isShutdownPending {
 return fmt.Errorf("shutdown in progress not allowed to shut down")
}

	if pushFront {
		h.orderedStopables = append([]Stopable{stopable}, h.orderedStopables...)
	} else {
		h.orderedStopables = append(h.orderedStopables, stopable)
	}
```